.github/workflows/ci.yaml

```diff
@@ -69,3 +71,6 @@ jobs:
       run: kubectl logs --tail 100 -ntesting -loperator.juju.is/name=spark
       if: failure()
 
+    - name: Run integration tests
+      run: tox -e integration
+
```

Comment:
This can move up and replace the existing `build and deploy`.  At a minimum, we don't want this below the `Get` blocks that have `if: failure()` - those blocks are to capture kubectl details if anything breaks so they come at the end

Also, these `failure()` log capture blocks get logs from the `testing` namespace, so make sure to use the `testing` model when running your CI so the logs are where they're expected.